saleor/graphql/account/mutations/base.py

```diff
@@ -93,19 +98,35 @@ class Arguments:
             required=True,
             description="Email of the user that will be used for password recovery.",
         )
+        redirect_url = graphene.String(
+            required=True,
+            description=(
+                "Url of storefront view which allow user reset password, "
+                "sent by email. Url in RFC 1808 format.",
+            ),
+        )
 
     class Meta:
         description = "Sends an email with the account password modification link."
 
     @classmethod
     def perform_mutation(cls, _root, info, **data):
         email = data["email"]
+        redirect_url = data["redirect_url"]
+
+        parsed_url = urlparse(redirect_url)
```

Comment:
It would be nice to extract this validation to a function, as we will want to validate the URL against `ALLOWED_STOREFRONT_HOSTS` in a few places. Not sure where we could put the function but maybe somewhere in `saleor/core`?